error: rules with generic types or `rule<_>` types cannot use #[cache] or #[cache_left_rec]
 --> tests/compile-fail/cache_with_args.rs:5:10
  |
5 |     rule ltarg<'a>() -> &'a str = { "" } //~ ERROR
  |          ^^^^^

error: rules with generic types or `rule<_>` types cannot use #[cache] or #[cache_left_rec]
 --> tests/compile-fail/cache_with_args.rs:8:10
  |
8 |     rule rulearg(r: rule<()>) -> &'a str = { "" } //~ ERROR
  |          ^^^^^^^
